src/ipc/windows/ipc.c

```diff
@@ -0,0 +1,226 @@
+#include "ipc.h"
+#include "defines/ipc_values.h"
+#include "defines/settings.h"
+#include "utils/ipUtils.h"
+#include "utils/logger.h"
+#include "utils/memory.h"
+#include "utils/oidc_error.h"
+#include "utils/printer.h"
+#include "utils/stringUtils.h"
+
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <winsock2.h>
+#include <process.h>
+
+#include "utils/registryConnector.h"
+#include "utils/file_io/file_io.h"
+
+
+
+/**
+ * @brief initializes a tcp socket on the loopback address (MSYS emulated)
+ * @param con, a pointer to the connection struct. The relevant fields will be
+ * initialized.
+ * @param remote, determines whether its remote communication or not (remote communication currently not supported for Windows)
+ */
+oidc_error_t ipc_client_init(struct connection* con, unsigned char remote) {
+  if (remote) {
+    logger(DEBUG, "Remote connections are currently not supported by windows oidc-agent library");
+    return OIDC_EERROR;
+  }
+	con->msys_server = secAlloc(sizeof(struct sockaddr_in));
+	con->sock = secAlloc(sizeof(SOCKET));
+
+  // Extract port and authorization string from MSYS socket file
+  char *socketpath = getRegistryValue(OIDC_SOCK_ENV_NAME);
+  int port;
+  sscanf(readFile(socketpath), "!<socket >%d s %X-%X-%X-%X", &port, &con->msys_secret[0], &con->msys_secret[1], &con->msys_secret[2], &con->msys_secret[3]);
```

Comment:
MSYS socket file has the structure used in sscanf. Should we define that as constant?

Reply:
As it is only used at this location and not multiple times, it is fine to leave it as it is